src/index.js

```diff
@@ -0,0 +1,457 @@
+        curUser.eventName = settings.preferences.eventName;
+        curUser.duration = settings.preferences.duration.toString();
+        curUser.firstShiftStart =
+          settings.preferences.firstShiftStart.toString();
+        curUser.secondShiftStart =
+          settings.preferences.secondShiftStart.toString();
+        curUser.wage = settings.preferences.wage;
+        curUser.calendarID = settings.preferences.calendarID;
+        await curUser.save();
+        if (settingsChanged) {
+          await bot.editMessageReplyMarkup({},
+            {
+              chat_id: CHAT_ID,
+              message_id: MESSAGE_ID,
+            });
+          return bot.sendMessage(chatId, 'Settings updated successfully✅')
+            .then(settingsChanged = false);
+        }
+        return bot.sendMessage(chatId, 'You didn\'t change anything😐');
+      case Constants.THIS_MONTH_SHIFTS:
+        [startDate, endDate] = [
+          startOfMonth(curDate),
+          endOfMonth(curDate),
+        ];
+        response = await showShifts(curUser.calendarID, {
+          startDate,
+          endDate,
+          eventName: curUser.eventName,
+        });
+        return bot.sendMessage(chatId, response, {
+          parse_mode: 'HTML',
+        });
+      case Constants.CURRENT_AND_PREVIOUS:
+        [startDate, endDate] = [
+          subMonths(startOfMonth(curDate), 1),
+          endOfMonth(curDate),
+        ];
+        response = await showShifts(curUser.calendarID, {
+          startDate,
+          endDate,
+          eventName: curUser.eventName,
+        });
+        return bot.sendMessage(chatId, response, {
+          parse_mode: 'HTML',
+        });
+      case Constants.ALL_SHIFTS_LIST:
+        [startDate, endDate] = [
+          subYears(curDate, 5),
+          addYears(curDate, 5),
+        ];
+        response = await showShifts(curUser.calendarID, {
+          startDate,
+          endDate,
+          eventName: curUser.eventName,
+        });
+        return bot.sendMessage(chatId, response, {
+          parse_mode: 'HTML',
+        });
+      default:
+        if (data.includes('EVENT_DATE')) {
+          const dateToTick = data.split(':')[1];
+          return bot.editMessageReplyMarkup({
+            inline_keyboard: InlineKeyboards.tickDay(
+              keyboard, dateToTick),
+          },
+          {
+            chat_id: chatId,
+            message_id: msgId,
+          });
+        }
+        if (data.includes('TIME_NOT_PICKED')) {
+          const dateToChange = data.split(':')[1];
+          return bot.editMessageReplyMarkup({
+            inline_keyboard: InlineKeyboards.changeShift(
+              keyboard, dateToChange),
+          },
+          {
+            chat_id: chatId,
+            message_id: msgId,
+          });
+        }
+        return bot.sendMessage(chatId, 'Invalid command, try again👀');
+    }
+  } catch (e) {
+    return bot.sendMessage(chatId, `Error occurred! ${e.message}`);
+  }
+});
```

Comment:
Blocker: a huge amount of copy-pasting methods/code.

Blocker: mix of a lot of levels of abstraction: from low-level components to high-level abstractions. It will be better to make services and use them in bot methods